bin/migration/summary.py

```diff
@@ -1,104 +1,163 @@
+            'booking_participant': 'booking_participant',
+            'court_region':'court_region',
+            'courtrooms':'courtrooms',
+            'role_permission':'role_permission',
+            'share_bookings':'share_bookings'
+        }
+
+    def fetch_source_data(self, query):
+        cursor = self.source_conn.cursor()
+        cursor.execute(query)
+        data = cursor.fetchone()[0]
+        cursor.close()
+        return data
+
+    def count_records_in_source_tables(self):
+        table_counts = {}
+
+        #Rooms
+        table_counts['rooms'] = self.fetch_source_data("SELECT COUNT(*) FROM public.rooms")
+        #Users
+        table_counts['users'] = self.fetch_source_data("SELECT COUNT(*) FROM public.users")
+        # Roles
+        table_counts['roles'] = self.fetch_source_data("SELECT COUNT(grouptype) FROM public.grouplist WHERE grouptype = 'Security'")
+        # Courts
+        table_counts['courts'] = self.fetch_source_data("SELECT COUNT(grouptype) FROM public.grouplist WHERE grouptype = 'Location'") + 1 # this adds 1 for the default court
+        #Regions
+        table_counts['regions'] = 12
+        # Portal Access
+        table_counts['portal_access'] = self.fetch_source_data("""
+                            SELECT COUNT(*) AS count_result FROM (
+	                            SELECT
+                                    u.userid,
+                                    u.status as active,
+                                    u.loginenabled as loginenabled,
+                                    u.invited as invited,
+                                    u.emailconfirmed as emailconfirmed,
+                                    MAX(ga.assigned) AS created,
+                                    MAX(ga.assignedby) AS createdby
+                                FROM public.users u
+                                JOIN public.groupassignments ga ON u.userid = ga.userid
+                                JOIN public.grouplist gl ON ga.groupid = gl.groupid
+                                WHERE gl.groupname = 'Level 3' OR u.invited ILIKE 'true'
+                                GROUP BY u.userid
+                            ) AS count""")        
+        # app access
+        table_counts['app_access'] = self.fetch_source_data("""
+                            SELECT COUNT(*) AS count_result FROM (
+                                SELECT 
+                                    u.userid,
+                                    COUNT(CASE WHEN gl.grouptype = 'Security' AND ga.groupid IS NOT NULL THEN 1 ELSE NULL END) AS role_id_count,
+                                    MAX(CASE WHEN gl.grouptype = 'Location' THEN ga.groupid ELSE NULL END) AS court_id,
+                                    MAX(u.status) AS active,
+                                    MAX(ga.assigned) AS created,
+                                    MAX(ga.assignedby) AS createdby,
+                                    MAX(ga.gaid) AS app_access_id
```

Comment:
```suggestion
                                    ga.gaid AS app_access_id
```
I don't think `MAX()` does anything here but I doubt it's causing any issues either.

Reply:
Agreed